src/Solvers/krylov.jl

```diff
@@ -0,0 +1,125 @@
+import Krylov
+import Krylov.FloatOrComplex
+
+using KernelAbstractions: @index, @kernel
+
+struct KrylovField{T, F <: AbstractField} <: AbstractVector{T}
+    field::F
+end
+
+function KrylovField(field::F) where F <: AbstractField
+    T = eltype(field)
+    return KrylovField{T,F}(field)
+end
+
+Base.size(kf::KrylovField) = size(kf.field)
+Base.length(kf::KrylovField) = length(kf.field)
+Base.getindex(kf::KrylovField, i::Int) = getindex(kf.field, i)
+
+Krylov.kdot(n::Int, x::KrylovField, y::KrylovField) = dot(x.field, y.field)
+
+Krylov.knorm(n::Int, x::KrylovField) = norm(x.field)
+
+function Krylov.kscal!(n::Int, s, x::KrylovField)
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _x[i,j,k] = s * _x[i,j,k]
+            end
+        end
+    end
+    return x
+end
+
+function Krylov.kaxpy!(n::Int, s, x::KrylovField, y::KrylovField)
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _y[i,j,k] += s * _x[i,j,k]
+            end
+        end
+    end
+    return y
+end
+
+function Krylov.kaxpby!(n::Int, s, x::KrylovField, t, y::KrylovField)
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                _y[i,j,k] = s * _x[i,j,k] + t * _y[i,j,k]
+            end
+        end
+    end
+    return y
+end
+
+Krylov.kcopy!(n::Int, y::KrylovField, x::KrylovField) = copyto!(y.field, x.field)
+Krylov.kfill!(x::KrylovField, val) = fill!(x.field, val)
+
+function Krylov.kref!(n::Integer, x::KrylovField, y::KrylovField, c, s)
+    mx, nx, kx = size(x.field)
+    _x = x.field
+    _y = y.field
+    for i = 1:mx
+        for j = 1:nx
+            for k = 1:kx
+                x_ijk = _x[i,j,k]
+                y_ijk = _y[i,j,k]
+                _x[i,j,k] = c       * x_ijk + s * y_ijk
+                _x[i,j,k] = conj(s) * x_ijk - c * y_ijk
```

Comment:
doesn't this overwrite the statement above?

Reply:
Line `77` should be `_y[i,j,k]`, it's a typo.